dask/dataframe/io/orc.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import, division, print_function
+
+from ..core import DataFrame
+from ...base import tokenize
+from ...bytes.core import get_fs_token_paths
+from ...utils import import_required
+from .utils import _get_pyarrow_dtypes, _meta_from_dtypes
+
+__all__ = ('read_orc',)
+
+
+def _read_orc_stripe(fs, path, stripe, columns=None):
+    """Pull out specific data from specific part of ORC file"""
+    orc = import_required('pyarrow.orc', 'Please install pyarrow >= 0.9.0')
+    with fs.open(path, 'rb') as f:
+        o = orc.ORCFile(f)
+        return o.read_stripe(stripe, columns).to_pandas()
+
+
+def read_orc(path, columns=None, storage_options=None):
+    """Read dataframe from ORC file(s)
+
+    Parameters
+    ----------
+    path: str or list(str)
+        Location of file(s), which can be a full URL with protocol specifier,
+        and may include glob character if a single string.
+    columns: None or list(str)
+        Columns to load. If None, loads all.
+    storage_options: None or dict
+        Further parameters to pass to the bytes backend.
+
+    Returns
+    -------
+    Dask.DataFrame (even if there is only one column)
+
+    Examples
+    --------
+    >>> d = read_orc('https://github.com/apache/orc/raw/'
+    ...              'master/examples/demo-11-zlib.orc')  # doctest: +SKIP
```

Comment:
It was fun to have this immediately accessible.  

Should I be concerned that a 100MB dataset had 385 partitions?  Is this typical granularity for ORC?  Also, I don't suppose we know of a nicer sample that has column names?

Just minor comments though.  Good to merge from my perspective.

Reply:
Each of the files in that demo set is atypical, demonstrating some feature or other. In this case, the file size is only 400kB, showing how effective packing/compression can be. I would think ORC ought to have large blocks, like parquet. Nevertheless, it's a good URL to point to for such examples, and one that's not likely to go away.

In any case, the tests don't use  that file because of github requests limits, but something more sensible with 8 partitions and two named columns (but this has a URL too unfriendly for a doc-string).